### PR TITLE
Specify type mapping explicitly in MemSQL

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

File: plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java [hunk 1]
```diff
@@ -160,10 +171,26 @@ protected boolean filterSchema(String schemaName)
...
+                return Optional.of(bigintColumnMapping());
+            case Types.REAL:
+                return Optional.of(realColumnMapping());
+            case Types.DOUBLE:
+                return Optional.of(doubleColumnMapping());
             case Types.CHAR:
             case Types.NCHAR: // TODO it it is dummy copied from StandardColumnMappings, verify if it is proper mapping
                 return Optional.of(defaultCharColumnMapping(typeHandle.getRequiredColumnSize(), false));
             case Types.VARCHAR:
+            case Types.LONGVARCHAR:
```
Comment (hashhar) [inline plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java:193]: Does the driver actually return LONGVARHCAR for some type?

File: plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java [hunk 2]
```diff
@@ -172,10 +199,28 @@ protected boolean filterSchema(String schemaName)
                     int scale = min(decimalDigits, getDecimalDefaultScale(session));
                     return Optional.of(decimalColumnMapping(createDecimalType(Decimals.MAX_PRECISION, scale), getDecimalRoundingMode(session)));
                 }
+                if (precision > Decimals.MAX_PRECISION) {
+                    break;
+                }
+                return Optional.of(decimalColumnMapping(createDecimalType(precision, max(decimalDigits, 0))));
+            case Types.BINARY:
+            case Types.VARBINARY:
+            case Types.LONGVARBINARY:
```
Comment (hashhar) [inline plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java:208]: Are all 3 returned by driver?

**Review by hashhar — COMMENTED** [review]
Can you submit this PR from a branch in trinodb/trino to make sure MemSQL tests run?

Looks mostly good otherwise.

File: plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java:193]: Yes, MemSQL `text`, `mediumtext` and `longtext` types return `LONGVARCHAR`.

File: plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java (hunk 2 above)
Comment (ebyhr) [reply plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java:208]: Yes, all 3 cases are required:
* binary → Types.BINARY
* varbinary(n), tinyblob → Types.VARBINARY
* blob, mediumblob, longblob, varbinary → Types.LONGVARBINARY

**Comment by ebyhr** [comment]
@hashhar MemSQL CI passed in https://github.com/trinodb/trino/runs/3729041636

**Review by hashhar — APPROVED** [review]

**Pull request merged**